ocaml/xapi/xapi_globs.ml

```diff
@@ -1255,6 +1257,12 @@ let other_options =
     , (fun () -> !extauth_ad_backend)
     , "Which AD backend used to talk to DC"
     )
+  ; ( "winbind_kerberos_encryption_type"
+    , Arg.Set_string winbind_kerberos_encryption_type
```

Comment:
I think this should use `Arg.String`, which calls a function that validates the string and stores a value of the proper type directly such that we never deal with the string.  Most of the problems above come from using `Set_string`.